Sources/Workspace/ManagedDependency.swift

```diff
@@ -14,195 +14,188 @@ import SourceControl
 import TSCBasic
 import TSCUtility
 
-/// An individual managed dependency.
-///
-/// Each dependency will have a checkout containing the sources at a
-/// particular revision, and may have an associated version.
-public class ManagedDependency {
-    /// Represents the state of the managed dependency.
-    public enum State: Equatable {
-
-        /// The dependency is a managed checkout.
-        case checkout(CheckoutState)
-
-        /// The dependency is in edited state.
-        ///
-        /// If the path is non-nil, the dependency is managed by a user and is
-        /// located at the path. In other words, this dependency is being used
-        /// for top of the tree style development.
-        case edited(AbsolutePath?)
+extension Workspace {
```

Comment:
`ManagedDependency` is already in the `Workspace` module. By the looks of it, any of the call sites changed in this PR could use the fully qualified `Workspace.ManagedDependency` without nesting it as a subtype of `Workspace` (`Workspace.Workspace`). Is there something I'm missing?

Reply:
my intent here was to make it less visible as a public type. ideally it would be internal / private type as it is an implementation detail, but right now that is not possible because the code is very leaky in this space. as such nesting it makes it a bit more hidden from users of the workspace module (not a top level type). open to better ideas on how to make this more of an implementation details / not leak

Reply:
Ah, got it. It looks like the snag is the associated `ManagedDependency.State` value in `ResolverPrecomputationError.differentRequirement`, and I don't see an obvious way around that.

My 2 cents: I don't have a good sense of the audience for the public API of the `SwiftPM` package, but I'm not immediately concerned about `ManagedDependency` being public. (`ManagedDependencies` I could see as an internal helper, though).

For now, I'm alright with this change if only because it creates parity with `Workspace.ManagedArtifact`.